clients/java/signalr/src/test/java/com/microsoft/aspnet/signalr/HubConnectionTest.java

```diff
@@ -29,6 +29,34 @@ public void checkHubConnectionState() throws Exception {
         assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
     }
 
+    @Test
+    public void transportCloseTriggersStopInHubConnection() throws Exception {
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+        hubConnection.start();
+        assertEquals(HubConnectionState.CONNECTED, hubConnection.getConnectionState());
+        mockTransport.stop();
+
+        assertEquals(HubConnectionState.DISCONNECTED, hubConnection.getConnectionState());
+    }
+
+    @Test
+    public void transportCloseWithErrorTriggersStopInHubConnection() throws Exception {
```

Comment:
Can we write a test that has a transport failure during `hubConnection.start()` and make sure `hubConnection.start()` wont hang